accel-pppd/iprange.c

```diff
@@ -54,12 +60,23 @@ static int parse_iprange(const char *str, struct iprange_t **range)
 	uint32_t ip_max;
 	uint8_t suffix;
 	size_t len;
+	char *l2tp_secret = NULL;
+	char *p1;
+	char *str;
+
+	str = opt->name;
 
 	if (!strcmp(str, "disable"))
 		goto disable;
 
 	ptr = str;
 
+	p1 = strstr(ptr, ",l2tp_secret");
+	if (p1) {
+		l2tp_secret=opt->val;
+		*p1 = '\0';
```

Comment:
Hi,

Would it be possible not to mutate the input string here? It might be a bit more difficult as the `u_parse_endstr` bit would need to be tweaked, but that would allow for a more robust and intuitive code : `opt` is clearly an input argument, and one does not expect its content to be changed by a function called `parse_iprange`.

Reply:
I think that's good feedback. One simple way to do it would just to temporarily strdup() the contents. Would that be sufficient?